hibernate-core/src/main/java/org/hibernate/query/internal/AbstractProducedQuery.java

```diff
@@ -1103,12 +1103,16 @@ else if ( QueryHints.HINT_PASS_DISTINCT_THROUGH.equals( hintName ) ) {
 		}
 
 		if ( !applied ) {
-			MSG_LOGGER.debugf( "Skipping unsupported query hint [%s]", hintName );
+			handleUnrecognizedHint(hintName, value);
 		}
 
 		return this;
 	}
 
+	protected void handleUnrecognizedHint(String hintName,Object value) {
```

Comment:
Please add a space after the comma (this is the one that made me comment as it's quite annoying).

Thanks!